### PR TITLE
Add accelerator support for Super Resolution example

### PR DESCRIPTION
**Pull request by framoncg**
Refactor Super Resolution example to utilize torch.accelerator API. 

Update super_resolution/main.py with accelerator flag
Update super_resolution/super_resolve.py with accelerator flag
Fix super_resolution/super_resolve.py for an issue when moving model and input to the desired device
Update README to match super_resolution/main.py and super_resolution/super_resolve.py flags as well as some wording
Update run_python_examples.sh to add new accelerator flag and include both  training and inference

### CODE REVIEW

File: super_resolution/README.md [hunk 1]
```diff
@@ -16,20 +16,23 @@ optional arguments:
   --testBatchSize       testing batch size
   --nEpochs             number of epochs to train for
   --lr                  Learning Rate. Default=0.01
-  --cuda                use cuda
-  --mps                 enable GPU on macOS
+  --accel               enables Accelerator use for training
```
Comment (jafraustro) [inline super_resolution/README.md:19]: "enables accelerator"

File: super_resolution/README.md [hunk 2]
```diff
@@ -16,20 +16,23 @@ optional arguments:
...
-`python main.py --upscale_factor 3 --batchSize 4 --testBatchSize 100 --nEpochs 30 --lr 0.001`
+```bash
+python main.py --accel --upscale_factor 3 --batchSize 4 --testBatchSize 100 --nEpochs 30 --lr 0.001
+```
 
 ### Super Resolve
 
-`python super_resolve.py --input_image dataset/BSDS300/images/test/16077.jpg --model model_epoch_500.pth --output_filename out.png`
+```bash
+python super_resolve.py --input_image dataset/BSDS300/images/test/16077.jpg --model model_epoch_30.pth --output_filename out.png
```
Comment (jafraustro) [inline super_resolution/README.md:37]: it seems that  `accel` argument is missing

File: super_resolution/main.py [hunk 3]
```diff
@@ -16,29 +16,22 @@
 parser.add_argument('--testBatchSize', type=int, default=10, help='testing batch size')
 parser.add_argument('--nEpochs', type=int, default=2, help='number of epochs to train for')
 parser.add_argument('--lr', type=float, default=0.01, help='Learning Rate. Default=0.01')
-parser.add_argument('--cuda', action='store_true', help='use cuda?')
-parser.add_argument('--mps', action='store_true', default=False, help='enables macOS GPU training')
+parser.add_argument('--accel', action='store_true', default=False, help='enables Accelerated training')
```
Comment (jafraustro) [inline super_resolution/main.py:19]: drop `default=False`, it is not necessary. 

`parser.add_argument('--accel', action='store_true', help='enables Accelerated training')`

Please review the other parser arguments

File: super_resolution/main.py [hunk 4]
```diff
@@ -16,29 +16,22 @@
...
-if opt.cuda:
-    device = torch.device("cuda")
-elif use_mps:
-    device = torch.device("mps")
+if opt.accel and torch.accelerator.is_available():
+    device = torch.accelerator.current_accelerator()
 else:
     device = torch.device("cpu")
 
+print("===> Setting device to {}".format(device))
```
Comment (jafraustro) [inline super_resolution/main.py:33]: Review the other examples. the message is a little different

File: super_resolution/super_resolve.py [hunk 5]
```diff
@@ -11,20 +11,23 @@
 parser.add_argument('--input_image', type=str, required=True, help='input image to use')
 parser.add_argument('--model', type=str, required=True, help='model file to use')
 parser.add_argument('--output_filename', type=str, help='where to save the output image')
-parser.add_argument('--cuda', action='store_true', help='use cuda')
+parser.add_argument('--accel', action='store_true', help='enable accelerated training')
```
Comment (jafraustro) [inline super_resolution/super_resolve.py:14]: "enables accelerator"

File: super_resolution/super_resolve.py [hunk 6]
```diff
@@ -11,20 +11,23 @@
...
 print(opt)
+if opt.accel:
+    device = torch.accelerator.current_accelerator()
+else:
+    device = torch.device("cpu")
+
 img = Image.open(opt.input_image).convert('YCbCr')
 y, cb, cr = img.split()
 
-model = torch.load(opt.model)
```
Comment (jafraustro) [inline super_resolution/super_resolve.py:21]: Is it not necessary to load the model?

**Review by jafraustro — CHANGES_REQUESTED** [review]

File: super_resolution/super_resolve.py (hunk 5 above)
Comment (framoncg) [reply super_resolution/super_resolve.py:14]: this will more likely get bumped, I need to fix the requirements and fix the functionality and then make the change for the api, but good catch
